src/Build.UnitTests/Globbing/MSBuildGlob_Tests.cs

```diff
@@ -326,20 +326,39 @@ public void GlobMatchingShouldWorkWithComplexRelativeLiterals()
         [InlineData(
             @"a/b\c",
             @"d/e\f/**\a.cs",
-            @"d\e/f\g/h\i/a.cs")]
+            @"d\e/f\g/h\i/a.cs",
+            @"d\e/f\", @"g/h\i/", @"a.cs")]
         [InlineData(
             @"a/b\c",
             @"d/e\f/*b*\*.cs",
-            @"d\e/f\abc/a.cs")]
+            @"d\e/f\abc/a.cs",
+            @"d\e/f\", @"abc/", @"a.cs")]
         [InlineData(
             @"a/b/\c",
             @"d/e\/*b*/\*.cs",
-            @"d\e\\abc/\a.cs")]
-        public void GlobMatchingIgnoresSlashOrientationAndRepetitions(string globRoot, string fileSpec, string stringToMatch)
+            @"d\e\\abc/\a.cs",
+            @"d\e\\", @"abc\\", @"a.cs")]
+        public void GlobMatchingIgnoresSlashOrientationAndRepetitions(string globRoot, string fileSpec, string stringToMatch,
+            string fixedDirectoryPart, string wildcardDirectoryPart, string filenamePart)
         {
             var glob = MSBuildGlob.Parse(globRoot, fileSpec);
 
             Assert.True(glob.IsMatch(stringToMatch));
+
+            MSBuildGlob.MatchInfoResult result = glob.MatchInfo(stringToMatch);
+            Assert.True(result.IsMatch);
+
+            string NormalizeSlashes(string path)
+            {
+                string normalizedPath = path.Replace(Path.DirectorySeparatorChar == '/' ? '\\' : '/', Path.DirectorySeparatorChar);
+                return NativeMethodsShared.IsWindows ? normalizedPath.Replace("\\\\", "\\") : normalizedPath;
```

Comment:
Is there any chance this would be a `\\server\share` style (UNC) path? If so, this would break it.

Reply:
Yep, this is test code. I just wanted to add something for `FixedDirectoryPartMatchGroup`, `WildcardDirectoryPartMatchGroup`, and `FilenamePartMatchGroup` because they had no coverage at all. The platform-specific path normalization logic makes this somewhat less elegant. Oh, and I made sure the test case passes without my changes, too.

Btw, I took a cursory look at CPS and project-system sources and don't see these properties used. Do you happen to know if they ever were used or maybe about some other component calling `IMSBuildGlob.MatchInfo`? Thank you!

Reply:
I see no reference to any of those three properties in the history of CPS or dotnet/project-system. I also see no reference to `MSBuildGlob.MatchInfoResult` (is that the type you meant?).

Reply:
Thank you for confirming! (Yes, that's the type, it is available only on `MSBuildGlob` not on `IMSBuildGlob`, sorry for the confusion.)